recipes/libev/all/conanfile.py

```diff
@@ -47,38 +53,51 @@ def layout(self):
         basic_layout(self, src_folder="src")
 
     def validate(self):
-        if is_msvc(self):
-            raise ConanInvalidConfiguration(f"{self.ref} is not supported by Visual Studio")
         if self.info.settings.os == "Windows" and self.info.options.shared:
             # libtool:   error: can't build i686-pc-mingw32 shared library unless -no-undefined is specified
             raise ConanInvalidConfiguration(f"{self.ref} can't be built as shared on Windows")
 
     def build_requirements(self):
-        if self._settings_build.os == "Windows":
+        if self._settings_build.os == "Windows" and not is_msvc(self):
             self.win_bash = True
             if not self.conf.get("tools.microsoft.bash:path", check_type=str):
                 self.tool_requires("msys2/cci.latest")
 
     def source(self):
         get(self, **self.conan_data["sources"][self.version],
             destination=self.source_folder, strip_root=True)
+        if not is_msvc(self):    
+            rm(self, "config.h", os.path.join(self.export_sources_folder, "src"))
 
     def generate(self):
-        env = VirtualBuildEnv(self)
-        env.generate()
-        tc = AutotoolsToolchain(self)
-        tc.generate()
+        if is_msvc(self):
+            tc = CMakeToolchain(self)
+            tc.generate()
+        else:    
+            env = VirtualBuildEnv(self)
+            env.generate()
+            tc = AutotoolsToolchain(self)
+            tc.generate()
 
     def build(self):
-        autotools = Autotools(self)
-        autotools.configure()
-        autotools.make()
+        if is_msvc(self):
+            apply_conandata_patches(self)
```

Comment:
my review was not addressed here, no conditional `apply_conandata_patches()` please

Reply:
i missed it, now it removed